Sources/Error Handling/PurchasesError.swift

```diff
@@ -39,7 +40,29 @@ extension PurchasesError {
     /// let errorCode = error as? ErrorCode
     /// ```
     var asPublicError: PublicError {
-        return NSError(domain: Self.errorDomain, code: self.errorCode, userInfo: self.userInfo)
+        let rootError: Error = self.rootError(from: self)
+        let rootNSError = rootError as NSError
+        var rootErrorInfo = [
+            "code": rootNSError.code,
+            "domain": rootNSError.domain,
+            "localizedDescription": rootNSError.localizedDescription,
+        ] as [String : Any]
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, visionOS 1.0, *) {
+            if let storeKitErrorInfo = self.getStoreKitErrorInfoIfAny(error: rootError) {
+                rootErrorInfo = rootErrorInfo.merging(["storeKitError": storeKitErrorInfo])
+            }
+        }
+        let userInfoToUse = self.userInfo.merging([ErrorDetails.rootErrorKey: rootErrorInfo])
+        return NSError(domain: Self.errorDomain, code: self.errorCode, userInfo: userInfoToUse)
```

Comment:
I am afraid we might lose some keys because of an unwanted overlap. I would just add a prefix like `rc_` to any new key we add to:
- be explicit on what we're adding
- avoid dropping useful data.

Reply:
Right, I added everything within a `rc_root_error` which I felt was enough as a namespace... I don't think it's worth adding to all nested properties then? Lmk if you think otherwise though!